docs/general/ddl/create-table.rst

```diff
@@ -147,15 +174,78 @@ Column names are restricted in terms of patterns:
     :ref:`subscript notation <sql_dql_object_arrays>` (e.g. ``col['id']``) are
     not allowed.
 
-Advanced use
-============
 
-Tables can be:
+.. _ddl-create-table-configuration:
+
+Table configuration
+===================
+
+You can configure tables in many different ways to take advantage of the range
+of functionality that CrateDB supports. For example:
+
+.. rst-class:: open
+
+- You can :ref:`partition <partitioned-tables>` a table into one or more
+  partitions with the :ref:`PARTITIONED BY <sql-create-table-partitioned-by>`
+  clause. You control how tables are partitioned by specifying one or more
+  :ref:`partition columns <gloss-partition-column>`. Each unique combination of
+  partition column values results in a new partition.
+
+  By partitioning a table, you can segment some :ref:`SQL statements
+  <gloss-statement>` (e.g., those used for :ref:`table optimization
+  <optimize>`, :ref:`import and export <importing_data>`, and :ref:`backup and
+  restore <snapshot-restore>`) by constraining them to one or more partitions.
+
+  .. SEEALSO::
+
+      `How-to guides: Tuning partitions for insert performance`_
+
+- You can split partitions into one or more :ref:`shards <ddl-sharding>` with
+  the :ref:`CLUSTERED BY <sql-create-table-clustered>` clause. You control how
+  CrateDB routes table rows to shards by specifying a :ref:`routing column
+  <gloss-routing-column>`.
+
+  You can use :ref:`cluster settings <conf_routing>` to configure how shards
+  are :ref:`balanced <conf-routing-allocation-balance>` across a cluster and
+  :ref:`allocated <ddl_shard_allocation>` to nodes (with :ref:`attribute-based
+  allocation <conf-routing-allocation-attributes>`, :ref:`disk-based allocation
+  <conf-routing-allocation-disk>`, or both).
+
+  By :ref:`distributing shards <concept-clustering>` across the cluster, you
+  can increase both `data availability`_ and `service resilience`_. With
```

Comment:
my thinking for this was that if a node holding one shard splits off from the cluster, you can still read the table, but some data will be missing. is that correct? if correct, my reasoning is:

- availability is improved in the case of unreliable networks and node failures (both pretty much guaranteed to happen at one point or another)
- this also makes the cluster more resilient to failures
- more shards on more nodes mean more opportunities for distributed query execution

the same would be true of replicas

if my understanding is correct and my reasoning is accurate (and useful for the reader), perhaps it would make sense to keep this para here, and then in the next bullet point about replication, I could make it more explicit that more replicas offer the same benefit?

related question:

- I added the point about read performance to the replicas bullet because I only saw that mentioned when reviewing the docs on replicas. do more shards (primary) also improve read performance? I suppose the would if "read performance" is a proxy for "increased parallelization via distributed query execution"

Reply:
> my thinking for this was that if a node holding one shard splits off from the cluster, you can still read the table, but some data will be missing. is that correct?

No that's not correct, the queries will fail.

> do more shards (primary) also improve read performance?

Not in general, it depends. E.g. multiple shards increases parallel execution but also requires merge phases. So using multiple shards on one node with multiple cpu's/core's probably perform better than using a single shard. With multiple shards across multiple nodes more variables come into account like more merges, network traffic, etc.

Reply:
aha. gotcha. thanks. I removed this para entirely

in the next bullet, on replication, I changed the final para to this:

> Replication can also improve read performance when it results in copies of the data being available on multiple nodes because this gives CrateDB more opportunities to parallelize query execution.

have I understood this (cause and effect) correctly?